package.json

```diff
@@ -3,6 +3,8 @@
   "version": "3.1.0",
   "description": "Core utilities for the react-navigation framework",
   "main": "dist/index.js",
+  "module": "src/index.js",
```

Comment:
I think we need a separate build for modules. If we expose the source for modules folder, people will have to transpile the library with babel. Which also means we can't guarantee that the lib will work with all babel configs, and we can't use typescript.

Reply:
> people will have to transpile the library with babel. 

This is what enables better treeshaking and dead code elimination. Expo will provided a babel config that can do this by default. 

> Which also means we can't guarantee that the lib will work with all babel configs.

Currently react navigation pulls in a _lot_ of code and very little can be shaken out, this really bogs down performance. [`babel-preset-expo`](https://github.com/expo/expo/blob/569595fe7b2a6c2f2063ce16ae6471699314bef5/packages/babel-preset-expo/index.js#L28-L130) (disregard the CJS part, that's coming out) uses `@babel/preset-env` which is far more efficient than a non-strict CJS based solution. 

> and we can't use typescript.

We can add TS back, I didn't realize it was as so important given nothing was actually typed.

Reply:
@EvanBacon so my hesitation is for few reasons:
1. It's not common on web to transpile node_modules, if we are adding webs support, it'll be nice to follow existing conventions
2. Webpack isn't as fast as metro, so transpiling node_modules has a visible negative impact on perf
3. It's not very user-friendly to transpile specific modules, you've to use some regex like `/node_modules[/\\](?!react-navigation-core)/`
4. For web, I'd like to use my own Babel config targeting browsers I support. I usually avoid using presets such as the react native preset or expo preset for web because I can't customize the transforms

> This is what enables better treeshaking and dead code elimination
Correct me if I'm wrong, but we only need to keep ES6 imports for tree-shaking, right? I'm not saying to transpile everything. For example, for `react-navigation-stack`, we selectively transpile some stuff: https://github.com/react-navigation/react-navigation-stack/blob/master/package.json#L19 and that's what I'm suggesting to do here

I'll leave it up to @brentvatne to decide, just my suggestions :)

BTW we also do same in paper, but we do want to transpile it before publishing so it's easier to consume.

Reply:
I removed commonjs babel plugin from the publish config so we can still take advantage of tree shaking and landed that, let me know if you think this will be problematic @satya164